src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -182,4 +198,156 @@ public static int parseHistory(String number) throws ParseException {
         }
         return Integer.parseInt(trimmedIndex);
     }
+
+    /**
+     * Parses {@code String trimmedArgs} into an {@code IdContainsKeywordsPredicate} and adds it to
+     * {@code ArrayList<Predicate<Person>> predicateList}. Leading and trailing whitespaces will be
+     * trimmed.
+     * @param predicateList contains {@code Predicate<Person>} that will be used to find employees that satisfy all the
+     *                      predicates.
+     * @param trimmedArgs keyword that is entered by the user.
+     * @throws ParseException if trimmedArgs is empty or if it contains '/' char.
+     */
+    public static void parseIdKeyword(ArrayList<Predicate<Person>> predicateList,
```

Comment:
Would it be better for the parse commands to be a pure function like the others ones in `ParserUtil`? For example the function would return `IdContainsKeyWordPredicate` instead of directly interacting with `predicateList`.

Reply:
oh yes I think that is possible!